nifpga/session.py

```diff
@@ -390,8 +413,250 @@ def read(self):
         """
         buf = self._ctype_type()
-        self._read_func(self._session, self._resource, buf, len(self))
-        return [bool(elem) if self._datatype is DataType.Bool else elem for elem in buf]
+        read_func = self._get_read_function()
+        read_func(self._session, self._resource, buf, len(self))
+        val = [bool(elem) if self._datatype is DataType.Bool else elem for elem in buf]
+        return val
+
+
+class _FxpRegister(_Register):
+    """
+    _FxpRegister is a private class that does all the work of converting the
+    LabVIEW fixed point data type into something more native to python. As a
+    user you will read and write to this register just as you would any other
+    register. Given the nature of fixed point there are a few caveats.
+
+    Just like the other registers, we do not support users creating their
+    instances of _FxpRegister, but after opening a session to a valid bitfile
+    the session.registers property will contain all registers fixed point
+    included.
+
+    Fixed point registers should be easily used from python with a few caveats:
+        1. Trying to write a value that does not conform to boundaries of the
+        defined register, will be coerced just as it would in labVIEW. Input
+        a value that needs to be coerced will result in a warning to the user.
+        A value is to be coerced if it is not a multiple of the delta value, or
+        if it exceeds the minimum or maximum values.
+        2. When writing to fixed point registers with the overflow status
+        enabled, you must pass in the tuple (overflow bool, value). Failing to
+        pass in the overflow boolean will raise an exception.
+    """
+    def __init__(self,
+                 session,
+                 nifpga,
+                 bitfile_register,
+                 base_address_on_device):
+        super(_FxpRegister, self).__init__(session,
+                                           nifpga,
+                                           bitfile_register,
+                                           base_address_on_device,
+                                           read_func=nifpga["ReadArray%s" % DataType.U32],
+                                           write_func=nifpga["WriteArray%s" % DataType.U32])
+        self._word_length = bitfile_register.word_length
+        self._integer_word_length = bitfile_register.integer_word_length
+        self._overflow_enabled = bitfile_register.overflow
+        self._signed = bitfile_register.signed
+        self._delta = self._calculate_delta()
+        self._minimum = self._calculate_minimum()
+        self._maximum = self._calculate_maximum()
+        self._transfer_len = self._calculate_transfer_len()
+        self._ctype_type = self._ctype_type * self._transfer_len
+
+    def _calculate_delta(self):
+        """ The value persisted in the bitfile for delta is not always
+        correct, therefore we must calculate it manually.
```

Comment:
You might also document what the delta is with regards to FXP or how its used.  

Same for min and max.  I assume they are the minimum and maximum values?

Reply:
Done